common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -68,14 +68,18 @@ def commit
       end
 
       # Returns the path to the cloned repo
-      def clone_repo_contents(target_directory: nil)
+      def clone_repo_contents
         @clone_repo_contents ||=
-          _clone_repo_contents(target_directory: target_directory)
+          _clone_repo_contents(target_directory: repo_contents_path)
```

Comment:
Looks like we are not be handing Octokit 404's when cloning the repo in `_clone_repo_contents`, might rarely end up there but wondering if we should handle it now that we'll be relying on it more widely. Think you want to rescue `SharedHelpers::HelperSubprocessFailed` and re-raise `Dependabot::RepoNotFound`